### PR TITLE
[IMP] mail: improve mail template usability

### PR DESCRIPTION
**Pull request by Bertrand2**
Allow our users to modify mail template more easily;
- make the list accessible from the settings
- give them a link to update relevant views to update header/footer
- make the list and form of templates more readable

task-2944770

### CODE REVIEW

File: addons/mail/models/mail_template.py [hunk 1]
```diff
@@ -85,6 +88,26 @@ def _compute_can_write(self):
...
+            raise NotImplementedError(_('Operation not supported'))
+        base_templates = self.env['mail.template'].search(
+            [('active', '=', True)]
+        ).filtered(lambda t: t.is_base_template)
+
+        # remove templates that we do not want to be put in evidence
+        base_templates_blacklist = base_templates.filtered(lambda t: not t.description)
+
+        search_bool = (operator == '=') == value
+        return [('id', 'in' if search_bool else 'not in', base_templates.ids), ('id', 'not in', base_templates_blacklist.ids)]
```
Comment (tde-banana-odoo) [inline addons/mail/models/mail_template.py:108]: Would be better to return a domain using fields on template model and avoid additional search queries, like (did not check it was the right one :D )
```
if operator == '=' and value or operation == '!=' and not value:
    return [('active', operator, value), ('description', '!=', False)]
return [('active', operator, value), ('description', '=', False)]
```

File: addons/mail/models/mail_template.py [hunk 2]
```diff
@@ -64,6 +66,7 @@ def default_get(self, fields):
     auto_delete = fields.Boolean(
         'Auto Delete', default=True,
         help="This option permanently removes any track of email after it's been sent, including from the Technical menu in the Settings, in order to preserve storage space of your Odoo database.")
+    description = fields.Text('Template description', translate=True, help="This field is used for internal description of the template's usage.")
```
Comment (tde-banana-odoo) [inline addons/mail/models/mail_template.py:69]: Just to keep fields ordered, I would define it under "name" as it is part of the template definition

**Review by tde-banana-odoo — COMMENTED** [review]
Zboing !

**Comment by Xavier-Do** [comment]
Upgrade exception [#141](https://runbot.odoo.com/web/#id=141&view_type=form&model=runbot.upgrade.exception) added.

Waiting the forward-port of this pr, the exception will be applied on all builds, and create an inconsistent state for migrations.
**Please forward port this asap up to master without change.** Exception should be forward-ported before the end of the week.

If you need to apply any change before it reaches master, please notify runbot team.

Details:
```
field:res.config.settings.secondary_color
field:res.config.settings.primary_color
field:mail.template.description
field:mail.template.is_base_template
field:mail.template.active
```
cc @KangOl @nseinlet

**Comment by tde-banana-odoo** [comment]
@robodoo r+ rebase-ff

**Comment by tde-banana-odoo** [comment]
@robodoo r+ rebase-ff

**Comment by tde-banana-odoo** [comment]
@robodoo r+ rebase-ff

**Pull request closed without merging**